conformance/utils/config/timeout.go

```diff
@@ -0,0 +1,129 @@
+limitations under the License.
+*/
+
+package config
+
+import "time"
+
+type TimeoutConfig struct {
+	// CreateTimeout represents the maximum time for a Kubernetes object to be created.
+	// Max value for conformant implementation: None
+	// Used by: MustApplyObjectsWithCleanup, MustApplyWithCleanup
+	CreateTimeout time.Duration
+
+	// DeleteTimeout represents the maximum time for a Kubernetes object to be deleted.
+	// Max value for conformant implementation: None
+	// Used by: MustApplyObjectsWithCleanup, MustApplyWithCleanup
+	DeleteTimeout time.Duration
+
+	// GatewayMustHaveAddress represents the maximum time for at least one IP Address has been set in the status of a Gateway.
+	// Max value for conformant implementation: None
+	// Used by: WaitForGatewayAddress
+	GatewayMustHaveAddress time.Duration
+
+	// GatewayStatusMustHaveListeners represents the maximum time for a Gateway to have listeners in status that match the expected listeners.
+	// Max value for conformant implementation: None
+	// Used by: GatewayStatusMustHaveListeners
+	GatewayStatusMustHaveListeners time.Duration
+
+	// GWCMustBeAccepted represents the maximum time for a GatewayClass to have an Accepted condition set to true.
+	// Max value for conformant implementation: None
+	// Used by: GWCMustBeAccepted
+	GWCMustBeAccepted time.Duration
+
+	// HTTPRouteMustNotHaveParents represents the maximum time for an HTTPRoute to have either no parents or a single parent that is not accepted.
+	// Max value for conformant implementation: None
+	// Used by: HTTPRouteMustHaveNoAcceptedParents
+	HTTPRouteMustNotHaveParents time.Duration
+
+	// HTTPRouteMustHaveCondition represents the maximum time for an HTTPRoute to have the supplied Condition.
+	// Max value for conformant implementation: None
+	// Used by: HTTPRouteMustHaveCondition
+	HTTPRouteMustHaveCondition time.Duration
+
+	// HTTPRouteMustHaveParents represents the maximum time for an HTTPRoute to have parents in status that match the expected parents.
+	// Max value for conformant implementation: None
+	// Used by: HTTPRouteMustHaveParents
+	HTTPRouteMustHaveParents time.Duration
+
+	// ManifestFetchTimeout represents the maximum time for getting content from a https:// URL.
+	// Max value for conformant implementation: None
+	// Used by: getContentsFromPathOrURL
+	ManifestFetchTimeout time.Duration
+
+	// NamespacesMustBeReady represents the maximum time for all Pods and Gateways in a namespaces to be marked as ready.
+	// Max value for conformant implementation: None
+	// Used by: NamespacesMustBeReady
+	NamespacesMustBeReady time.Duration
+
+	// RequestTimeout represents the maximum time for making an HTTP Request or fetching a Kubernetes object.
+	// Max value for conformant implementation: None
+	// Used by: CaptureRoundTrip, GWCMustBeAccepted, NamespacesMustBeReady, WaitForGatewayAddress, HTTPRouteMustHaveNoAcceptedParents, HTTPRouteMustHaveParents, GatewayStatusMustHaveListeners, HTTPRouteMustHaveCondition
```

Comment:
I know I originally suggested this, but given how many functions use these timeouts, it will likely be difficult/impossible to keep these in sync as the codebase evolves. It's probably better just to leave the "Used by" lines out for all of these comments.

Reply:
Done.